src/system/signal/stream.rs

```diff
@@ -97,6 +100,8 @@ pub(crate) fn register_handlers<const N: usize>(
             })?;
     }
 
+    // SAFETY: if the above for-loop has terminated, every handler will have
+    // been written to via "MaybeUnit::new", and so is initialized.
     Ok(handlers.map(|(_, handler)| unsafe { handler.assume_init() }))
```

Comment:
Is it important for `SignalHandler::register` to not run on any other signals once an error has occured? If not it seems like it should be possible to rewrite this to avoid `MaybeUninit` entirely by doing the `SignalHandler::register` in the `<[SignalNumber; N]>::map()` call of this function, then iterate to check if any contains an error and return it and finally at this line use `handlers.map(|(_, handler)| handler.unwrap())`. In the future once `try_map` gets stabilized it can be switched to a single pass.

Reply:
I don't think that nuance is important in that error scenario and avoiding a `MaybeUninit` would be nice. @pvdrz?

Reply:
I think it is not important to avoid calling `register` if a previous error occurred

Reply:
Using the `map` route as suggested isn't actually that easy since the `Result` type then is stored/borrowed and its error-type is neither Clone nor Copy.